src/app/containers/MostRead/Canonical/index.jsx

```diff
@@ -1,39 +1,75 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useContext } from 'react';
 import 'isomorphic-fetch';
 import { string } from 'prop-types';
+import { MostRead } from '@bbc/psammead-most-read';
+import { ServiceContext } from '#contexts/ServiceContext';
 import webLogger from '#lib/logger.web';
+import { mostReadRecordIsFresh } from '../utilities';
+import LastUpdated from './lastUpdated';
 
 const logger = webLogger();
 
 const CanonicalMostRead = ({ endpoint }) => {
-  const [promos, setPromo] = useState([]);
-  const [data, setData] = useState({});
-
-  const handleResponse = async response => {
-    const mostReadData = await response.json();
-    setPromo(mostReadData.records.slice(0, 10));
-    setData(mostReadData);
-  };
+  const [items, setItems] = useState([]);
+  const {
+    service,
+    script,
+    dir,
+    datetimeLocale,
+    mostRead: { header, lastUpdated, numberOfItems },
+  } = useContext(ServiceContext);
 
   useEffect(() => {
+    const handleResponse = async response => {
+      const mostReadData = await response.json();
+
+      // do not show most read if lastRecordUpdated is greater than 35min
+      if (!mostReadRecordIsFresh(mostReadData.lastRecordTimeStamp)) {
+        return;
+      }
+      const mostReadItems = [];
+      mostReadData.records
+        .slice(0, numberOfItems)
+        .forEach(({ id, promo: { headlines, locators, timestamp } }) => {
+          mostReadItems.push({
+            id,
+            title: headlines.shortHeadline,
+            href: locators.assetUri,
+            timestamp: (
+              <LastUpdated
+                prefix={lastUpdated}
+                script={script}
+                service={service}
+                timestamp={timestamp}
+                locale={datetimeLocale}
+              />
+            ),
+          });
+        });
+      setItems(mostReadItems);
+    };
+
     const fetchMostReadData = pathname =>
       fetch(pathname, { mode: 'no-cors' })
         .then(handleResponse)
         .catch(e => logger.error(`HTTP Error: "${e}"`));
 
     fetchMostReadData(endpoint);
-  }, [endpoint]);
+  }, [endpoint, numberOfItems, datetimeLocale, lastUpdated, script, service]);
 
   return (
     <>
-      <p>Last Updated: {data.lastRecordTimeStamp}</p>
-      {promos.map(({ id, promo: { timestamp, headlines, locators } }) => (
-        <ul key={id}>
-          <li>{timestamp}</li>
-          <li>{headlines.headline}</li>
-          <li>{locators.assetUri}</li>
-        </ul>
-      ))}
+      {/* Render nothing when items is empty */}
+      {items.length !== 0 ? (
+        <MostRead
+          items={items}
+          header={header}
+          service={service}
+          script={script}
+          dir={dir}
+          labelId="Most-Read"
```

Comment:
NOTE: We need to ask if we want the section label localised or not

Reply:
if so, we have can use the header translations, but I suspect we need to use `-` instead of a space between the words

Reply:
This `labelId` prop is only used for the `MostReadSection` component. Since this is only used via `aria-labelledby` and isn't actually read out, this doesn't need to be localised and can be whatever we want.